Code Refactor

albertomurillo opened the pull request:
Fix https://github.com/cri-o/cri-o-ansible/issues/7

	Install packages for Fedora/RedHat/CentOS in one task 
	Use update_cache option from apt module
	Insert blank lines between tasks for better readability.
	Shorten comparison of ansible_distribution variable
	Install git on Ubuntu
	Add *.retry to .gitignore
	Check if go is already installed
	Some idempotency fixes

Commit:
Install packages for Fedora/RedHat/CentOS in one task

Fedora, RedHat and CentOS requires the same packages so it make
sense to group them in a single task delegating the package module
to choose from dnf/yum.

Also sort package names alphabetically.

Commit:
Use update_cache option from apt module

Do not run 'command: "apt-get update"' task whic is not idempotent.
Use the update_cache option from apt module instead.

Also sort package names alphabetically.

Commit:
Insert blank lines between tasks for better readability.

Commit:
Shorten comparison of ansible_distribution variable

Commit:
Install git on Ubuntu

runcom commented on cri-o.yml:
```diff
@@ -42,6 +42,7 @@
         - build-essential
         - curl
         - gawk
+        - git
```
could you elaborate on why this is needed?

albertomurillo replied:
Well there are tasks to clone git repositories and my Ubuntu installation didn't had git by default.

runcom commented:
@albertomurillo thanks for this! I'm about to merge #5 though so that will cause a rebase for sure :(

runcom commented:
@albertomurillo could you rebase and I'll give this a try asap?

Commit:
Add *.retry to .gitignore

Commit:
Check if go is already installed

Altough the 'install Golang upstream' task is idempotent it
downloads the package everytime.

This fix installs golang from upstream only when go is not already
installed or has a different version.

runcom commented:
the play is stuck at:
```
 ______________________________________
< TASK [Update all packages on Fedora] >
 --------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

skipping: [192.168.122.53]
skipping: [192.168.122.241]

```
it's supposed to `dnf update` in Fedora, but if I ssh in the machine I can `dnf update` myself, meaning no dnf process is already running in there :confused:

runcom commented:
So I provisioned a 3 node cluster with this playbook in this PR but somehow `kubeadm init` doesn't work anymore, I'm re-trying another cluster with master to understand if this PR is the issue.

runcom commented:
so it does work with master, I'm retrying one more time with this PR before saying it's an issue with this PR :(

runcom commented on cri-o.yml:
```diff
@@ -149,69 +147,75 @@
+      sysctl:
+        name: net/bridge/bridge-nf-call-iptables
+        value: 1
+        sysctl_file: /etc/sysctl.conf
```
https://docs.ansible.com/ansible/sysctl_module.html can drop this as it's the default

runcom commented on cri-o.yml:
```diff
@@ -149,69 +147,75 @@
-    - name: reload sysctl
-      command: "sysctl -p"
+      sysctl:
+        name: net/bridge/bridge-nf-call-iptables
```
this could be `net.bridge.bridge-nf-call-iptables` following https://docs.ansible.com/ansible/sysctl_module.html

runcom commented:
@albertomurillo have you tested if this is working on Fedora? I couldn't get it work as per our previous IRC chat :(

albertomurillo commented:
@runcom no but it didnt work either with master... I think is an issue with not restarting crio and kubelet services after `/etc/crio/crio.conf` and `/etc/systemd/system/kubelet.service.d/0-crio.conf` are changed

runcom commented:
> @runcom no but it didnt work either with master... I think is an issue with not restarting crio and kubelet services after /etc/crio/crio.conf and /etc/systemd/system/kubelet.service.d/0-crio.conf are changed

may be right, actually, we should do a daemon-reload after we install the systemd drop in. WRT crio.conf I think crio isn't started when we edit it, so that's probably ok.

albertomurillo commented:
I have the change to reload daemons ready since yesterday im testing them right now but my internet is a bit slow :P

Commit:
Reload crio daemon after configuration file changes

Signed-off-by: Alberto Murillo Silva <alberto.murillo.silva@intel.com>

Commit:
Replace calls to command with calls to proper ansible modules

Signed-off-by: Alberto Murillo Silva <alberto.murillo.silva@intel.com>

Commit:
Fix install policy.json task

task: install policy.json in Ubuntu
was running in all distros.

Fixes #7

albertomurillo commented:
This is ready now.

runcom commented on cri-o.yml:
```diff
@@ -209,9 +212,12 @@
+        content: |
+                  [Service]
+                  Environment="KUBELET_EXTRA_ARGS=--enable-cri=true --container-runtime=remote --runtime-request-timeout=15m --image-service-endpoint /var/run/crio.sock --container-runtime-endpoint /var/run/crio.sock"
+      notify: restart kubelet
```
tihs isn't needed, `kubeadm` is taking care of this

albertomurillo replied:
done

runcom commented:
I still can't make this to work on Fedora 25 :confused: 

```
[certificates] Generated service account token signing key and public key.
[certificates] Generated front-proxy CA certificate and key.
[certificates] Generated front-proxy client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
[apiclient] Created API client, waiting for the control plane to become ready
[apiclient] All control plane components are healthy after 145.514402 seconds
[apiclient] Waiting for at least one node to register
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
```

albertomurillo commented:
Im splitting this PR into smaller ones.
